upgrader.go

```diff
@@ -68,7 +68,7 @@ func (u *Upgrader) Upgrade(ctx context.Context, t transport.Transport, maconn ma
 	if dir == network.DirOutbound && p == "" {
 		return nil, ErrNilPeer
 	}
-	var stat network.Stat
+	var stat network.ConnStats
```

Comment:
i dont think we are counting streams in the underlying conn, so we should be fine with the early copy.